tests/backup/backup_test.go

```diff
@@ -1828,186 +1828,184 @@ func killStork() {
-					SkipClusterScopedObjects: true,
-				})
+		Step("Creating cloud credentials", func() {
+			log.InfoD("Creating cloud credentials")
+			for _, provider := range providers {
+				CredName := fmt.Sprintf("%s-%s", "cred", provider)
+				CloudCredUID = uuid.New()
+				CloudCredUIDMap[CloudCredUID] = CredName
+				CreateCloudCredential(provider, CredName, CloudCredUID, orgID)
 			}
 		})
 
-		// Change namespaces to restored apps only after backed up apps are cleaned up
-		// to avoid switching back namespaces to backup namespaces
-		Step(fmt.Sprintf("Validate Restored applications"), func() {
-			destClusterConfigPath, err := GetDestinationClusterConfigPath()
-			Expect(err).NotTo(HaveOccurred(),
-				fmt.Sprintf("Failed to get kubeconfig path for destination cluster. Error: [%v]", err))
-
-			SetClusterContext(destClusterConfigPath)
+		Step("Register cluster for backup", func() {
+			CreateSourceAndDestClusters(orgID, "", "", ctx)
+			clusterStatus, clusterUid = Inst().Backup.RegisterBackupCluster(orgID, SourceClusterName, "")
+			dash.VerifyFatal(clusterStatus, api.ClusterInfo_StatusInfo_Online, "Verifying backup cluster")
+		})
 
-			for _, ctx := range contexts {
-				err = Inst().S.WaitForRunning(ctx, defaultTimeout, defaultRetryInterval)
-				Expect(err).NotTo(HaveOccurred())
+		Step("Creating backup location", func() {
+			log.InfoD("Creating backup location")
+			bucketNames := getBucketName()
+			for _, provider := range providers {
+				CredName := fmt.Sprintf("%s-%s", "cred", provider)
+				bucketName := fmt.Sprintf("%s-%s", provider, bucketNames[0])
+				backupLocation = fmt.Sprintf("%s-%s", provider, bucketNames[0])
+				backupLocationUID = uuid.New()
+				CreateBackupLocation(provider, backupLocation, backupLocationUID, CredName, CloudCredUID,
+					bucketName, orgID, "")
 			}
-			// TODO: Restored PVCs are created by stork-snapshot StorageClass
-			// And not by respective app's StorageClass. Need to fix below function
-			ValidateRestoredApplications(contexts, volumeParams)
 		})
 
-		Step("teardown all restored apps", func() {
-			for _, ctx := range contexts {
-				TearDownContext(ctx, nil)
+		Step("Start backup of application to bucket", func() {
+			for _, namespace := range bkpNamespaces {
+				ctx, err := backup.GetAdminCtxFromSecret()
+				dash.VerifyFatal(err, nil, "Getting context")
+				preRuleUid, _ := Inst().Backup.GetRuleUid(orgID, ctx, preRuleNameList[0])
+				postRuleUid, _ := Inst().Backup.GetRuleUid(orgID, ctx, postRuleNameList[0])
+				backupName := fmt.Sprintf("%s-%s-%s", BackupNamePrefix, namespace, backupLocationName)
+				CreateBackupWithoutCheck(backupName, SourceClusterName, backupLocation, backupLocationUID, []string{namespace},
+					labelSelectors, orgID, clusterUid, preRuleNameList[0], preRuleUid, postRuleNameList[0], postRuleUid, ctx)
 			}
 		})
 
-		Step("teardown backup objects", func() {
-			TearDownBackupRestore(bkpNamespaces, bkpNamespaces)
+		storageNodes := node.GetWorkerNodes()
```

Comment:
I hope this operation does not take too much time, because if the backup process is completed by the time you get the worker nodes, then it would defeat the purpose of restarting px while backup is in progress

Reply:
no it is the data collected as part of before suite